src/tigerbeetle/cli.zig

```diff
@@ -18,80 +19,110 @@ const StateMachine = vsr.state_machine.StateMachineType(
+        \\           The --cluster and --replica arguments are required.
+        \\           Each TigerBeetle replica must have its own data file.
+        \\
+        \\  start    Run a TigerBeetle replica from the data file at <path>.
+        \\
+        \\  version  Print the TigerBeetle build version and the compile-time config values.
+        \\
+        \\Options:
+        \\
+        \\  -h, --help
+        \\        Print this help message and exit.
+        \\
+        \\  --cluster=<integer>
+        \\        Set the cluster ID to the provided 32-bit unsigned integer.
+        \\
+        \\  --replica=<index>
+        \\        Set the zero-based index that will be used for the replica process.
+        \\        An index greater than or equal to "replica-count" makes the replica a standby.
+        \\        The value of this argument will be interpreted as an index into the --addresses array.
+        \\
+        \\  --replica-count=<integer>
+        \\        Set the number of replicas participating in replication.
+        \\
+        \\  --addresses=<addresses>
+        \\        Set the addresses of all replicas in the cluster.
+        \\        Accepts a comma-separated list of IPv4 addresses with port numbers.
+        \\        Either the IPv4 address or port number (but not both) may be omitted,
+        \\        in which case a default of {[default_address]s} or {[default_port]d}
+        \\        will be used.
+        \\        "addresses[i]" corresponds to replica "i".
+        \\
+        \\  --cache-grid=<size><KB|MB|GB>
+        \\        Set the grid cache size. The grid cache acts like a page cache for TigerBeetle,
+        \\        and should be set as large as possible.
+        \\        On a machine running only TigerBeetle, this is somewhere around
+        \\        (Total RAM) - 3GB (TigerBeetle) - 1GB (System), eg 12GB for a 16GB machine.
+        \\        Defaults to 1GB.
+        \\
+        \\  --verbose
+        \\        Print compile-time configuration along with the build version.
+        \\
+        \\Examples:
+        \\
+        \\  tigerbeetle format --cluster=0 --replica=0 --replica-count=3 0_0.tigerbeetle
+        \\  tigerbeetle format --cluster=0 --replica=1 --replica-count=3 0_1.tigerbeetle
+        \\  tigerbeetle format --cluster=0 --replica=2 --replica-count=3 0_2.tigerbeetle
+        \\
+        \\  tigerbeetle start --addresses=127.0.0.1:3003,127.0.0.1:3001,127.0.0.1:3002 0_0.tigerbeetle
+        \\  tigerbeetle start --addresses=3003,3001,3002 0_1.tigerbeetle
+        \\  tigerbeetle start --addresses=3003,3001,3002 0_2.tigerbeetle
+        \\
+        \\  tigerbeetle start --addresses=192.168.0.1,192.168.0.2,192.168.0.3 0_0.tigerbeetle
+        \\
+        \\  tigerbeetle version --verbose
+        \\
+    , .{
+        .default_address = constants.address,
+        .default_port = constants.port,
+    });
+};
 
 pub const Command = union(enum) {
```

Comment:
Note that we have both `CliArgs` with "syntactically valid" arguments, and `Command` with semantically-validated ones. I actually like this overall pattern when doing any kind of parsing/deserialization --- rather than trying to cram all the validation into auto-generatied code, do only shallow syntax validation when parsing from a string to a type, and then have a second layer of manually written dumb validation which transforms between two types (but doesn't have to do parsing itself).